### DISCUSSION TITLE
Build error when importing an existing pom less tycho build

### DISCUSSION THREAD

**Discussion by moley** [eclipse-m2e/m2e-core#1054] [Q&A]
Hi,

I have a pom less tycho build, which is successful on commandline (both ./mvnw eclipse:eclipse and ./mvnw clean verify). 
After importing it into eclipse via "Import -> Existing Maven Projects..." I get the projects imported and the sourcefolder e.g. is detected correctly, but the projects are not marked as java projects nor is the java builder configured. I recieve as well errors like "Project x is missing required Java project y" although they are navigatable via manifest editor. 
I noticed that .project file is available for every project ( even the grouping folders "bundles", "features",.....) but the .classpath file is only available within projects under "tests". Can you please point me in the right direction how to solve. 

Is calling the named tasks enough? What is the default way to import an existing pomless tycho build into eclipse?

**Comment by HannesWell** [comment]
In general calling the `eclipse:eclipse` goal should not be necessary to work with Maven projects in Eclipse, if you have m2e installed.

Recently the support for Tycho projects was improved but those changes have not yet been released (but will be soon to be in the December SimRel). So maybe you want to try the snapshot at:
https://download.eclipse.org/technology/m2e/snapshots/2.1.0/

But what I wonder how your project looks exactly.
You say you have a pom-less build (i.e. no pom.xml in the project), but also don't maintain a `.project` and `.classpath` file manually respectively don't have that in your git repo. Pomless is usually used for 'pure' Eclipse PDE projects that usually have a `META-INF/MANIFEST.MF`, a `.project`, `.classpath` and `build.properties` file, that are also maintained directly/manually.

Therefore can you please share a minimal example (how it is checked into you version control) and the steps to set it up?
But as said before invoking the `eclipse:eclipse` goal should not be necessary. If your case is not too edgy, eventually we/you can think about enhancing M2E that it works out of the box (your minimal example should then serve as test case).
